backend/backend-python/model_loader.py

```
@@ -69,31 +69,63 @@ def load_model(
-                        if tensor_data.dtype != param.dtype:
-                            tensor_data = tensor_data.to(dtype=param.dtype)
-                        if tensor_data.device != param.device:
-                            tensor_data = tensor_data.to(device=param.device)
+                        # Convert to torch.Tensor if needed and check dtype/device compatibility
+                        tensor_dtype = getattr(tensor_data, "dtype", None)
+                        if (
+                            hasattr(tensor_data, "dtype")
+                            and tensor_dtype != param.dtype
+                        ):
+                            if hasattr(tensor_data, "to"):
+                                tensor_data = getattr(tensor_data, "to")(
```

Ruff (0.13.1)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)